enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,269 @@
+---
+title: coreos-image-in-release
+authors:
+  - "@zaneb"
+reviewers:
+  - "@hardys"
+  - "@dtantsur"
+  - "@elfosardo"
+  - "@sadasu"
+  - "@kirankt"
+  - "@asalkeld"
+  - "@cgwalters"
+  - "@aravindhp"
+  - "@jlebon"
+  - "@sosiouxme"
+  - "@dhellmann"
+  - "@sdodson"
+  - "@LorbusChris"
+approvers:
+  - "@hardys"
+  - "@aravindhp"
+creation-date: 2021-09-22
+last-updated: 2021-09-22
+status: implementable
+see-also:
+  - "/enhancements/coreos-bootimages.md"
+---
+
+# Include the CoreOS image in the release for baremetal
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The baremetal platform is switching from the OpenStack QCOW2 CoreOS image to
+the live ISO (as also used for UPI). To ensure that existing disconnected
+clusters can update to this, the ISO image will be included in the release
+payload. This will be balanced out by removing the RHEL image currently shipped
+in the release payload.
+
+## Motivation
+
+Currently, the deploy disk image (i.e. the image running IPA -
+`ironic-python-agent`) is a RHEL kernel plus initrd that is installed (from an
+RPM) into the `ironic-ipa-downloader` container image, which in turn is part of
+the OpenShift release payload. When the metal3 Pod starts up, the disk image is
+copied from the container to a HostPath volume whence it is available to
+Ironic.
+
+The target OS disk image is a separate CoreOS QCOW2 image. The URL for this is
+known by the installer. It points to the public Internet by default and may be
+customised by the user to allow disconnected installs. The URL is stored in the
+Provisioning CR at install time and never updated automatically. The image
+itself is downloaded once and permanently cached on all of the master nodes.
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
```

Comment:
this is capped the "the time it takes to download and setup a completely unrelated OS image", right?  E.g. if you install with release A, and later by release B there is now nothing from A that survives, the A->B new-node time is as slow as it gets, and a later release C will have a similar A->C new-node time (assuming C is around the same size as B)?  It's still not great, and it's enough to motivate this enhancement, but maybe worth rewording to something like:

> ... new Machine will take longer (and more bandwidth) to join as the cluster's desired OS diverges from the boot image, capping out when the desired OS is completely different from the boot image.